next/pages/_app.tsx

```diff
@@ -15,21 +21,27 @@ import {
 import { CommonPageProps } from "@/lib/get-common-page-props";
 
 interface PageProps extends CommonPageProps {
+  dehydratedState: any;
   languageLinks?: LanguageLinks;
   session?: Session;
 }
 
 function App({ Component, pageProps }: AppProps<PageProps>) {
+  const [queryClient] = React.useState(() => new QueryClient());
   const { menus, languageLinks, session, ...restPageProps } = pageProps;
   return (
     <SessionProvider session={session}>
-      <Fonts>
-        <LanguageLinksProvider languageLinks={languageLinks}>
-          <Layout menus={menus}>
-            <Component {...restPageProps} />
-          </Layout>
-        </LanguageLinksProvider>
-      </Fonts>
+      <QueryClientProvider client={queryClient}>
+        <Fonts>
+          <LanguageLinksProvider languageLinks={languageLinks}>
+            <Layout menus={menus}>
+              <Hydrate state={pageProps.dehydratedState}>
```

Comment:
I am not 100% sure if this is really needed if we use only client side queries?

Reply:
I'm not sure either. I suppose if we don't notice any difference then we can just remove it, since adding it there implies it's needed. But it depends what role react-query will eventually play in this project, maybe we will need to add it soon enough anyway?

Reply:
Looks like we don't need the Hydrate part, I have removed it.